src/adapters/chineseloc/chinese-source/chinese-source.js

```diff
@@ -0,0 +1,385 @@
+          result = ChineseSource.findWord(cpWord, dWordIndexSimp, dWordDict)
+        }
+      }
+
+      if (result) {
+        result.forEach(resItem => {
+          rs[count++] = resItem
+          rs[count - 1].format = format
+
+          ChineseSource.formatDictionaryEntry(rs[count - 1])
+          ChineseSource.formatCharacterInfo(rs[count - 1], dHanziDict)
+          rs[count - 1].pinyin = ChineseSource.formatPinyin(rs[count - 1].pinyin)
+        })
+      }
+      cpWord = cpWord.substring(0, cpWord.length - 1)
+    }
+
+    return rs
+  }
+
+  static formatDictionaryEntry (resItem) {
+    if (resItem.format === 'trad') {
+      resItem.dictEntry = resItem.word.split(' ')[0]
+    } else if (resItem.format === 'simp') {
+      resItem.dictEntry = resItem.word.split(' ')[1]
+    }
+  }
+
+  static formatCharacterInfo (resItem) {
+    // console.info('formatCharacterInfo dHanziDict - ', dHanziDict)
+    const freqName =
+      [
+        'least frequent',
+        'less frequent',
+        'moderately frequent',
+        'more frequent',
+        'most frequent'
+      ]
+
+    let unicode = ChineseSource.unicodeInfo(resItem.dictEntry)
+    let hanziDatElement = dHanziDict.get(unicode)
+    // console.info('hanziDatElement - ', hanziDatElement)
+
+    if (hanziDatElement) {
+      // console.info('hanziDatElement inside')
+
+      if (hanziDatElement.kMandarin) {
+        resItem.mandarin = 'mandarin - ' + ChineseSource.formatPinyin(hanziDatElement.kMandarin.toLowerCase())
+      }
+      if (hanziDatElement.kDefinition) {
+        resItem.definition = hanziDatElement.kDefinition
+      }
+      if (hanziDatElement.kCantonese) {
+        resItem.cantonese = 'cantonese - ' + hanziDatElement.kCantonese
+      }
+      if (hanziDatElement.kTang) {
+        resItem.tang = 'tang - ' + hanziDatElement.kTang
+      }
+      if (hanziDatElement.kFrequency) {
+        resItem.frequency = freqName[hanziDatElement.kFrequency - 1]
+      }
+      if (hanziDatElement.kRSUnicode) {
+        resItem.unicode = ChineseSource.radicalIndexToChar(hanziDatElement.kRSUnicode.split('.')[0])
+      }
+    }
+  }
+
+  static unicodeInfo (word) {
+    const hex = '0123456789ABCDEF'
+    const u = word.charCodeAt(0)
+    return 'U+' +
+        hex[(u >>> 12) & 15] +
+        hex[(u >>> 8) & 15] +
+        hex[(u >>> 4) & 15] +
+        hex[u & 15]
+  }
+
+  static formatPinyin (aPinyin) {
+    // console.info('formatPinyin - ', aPinyin)
+    // pinyin info
+    const _a = ['\u0101', '\u00E1', '\u01CE', '\u00E0', 'a']
+    const _e = ['\u0113', '\u00E9', '\u011B', '\u00E8', 'e']
+    const _i = ['\u012B', '\u00ED', '\u01D0', '\u00EC', 'i']
+    const _o = ['\u014D', '\u00F3', '\u01D2', '\u00F2', 'o']
+    const _u = ['\u016B', '\u00FA', '\u01D4', '\u00F9', 'u']
+    const _v = ['\u01D6', '\u01D8', '\u01DA', '\u01DC', '\u00FC']
+
+    aPinyin = aPinyin.split(/(\d)/)
+
+    let formatedPinyin = []
+
+    let toneFormat = {
+      1: 0, 2: 1, 3: 2, 4: 3

```

Comment:
good catch - there are only 4 tones in Mandarin, but I think in pinyin notation 0 and 5 might both be used for neutral tone  (see https://en.wikipedia.org/wiki/Pinyin#Numerals_in_place_of_tone_marks) so the code does need to be updated to account for both possibilities.

Reply:
may be we should update it in the final chinese service? or should I change anything in this temporary implementation?

Reply:
I think we can change it in the Chinese source file during conversion, we just need to decide which value will we use inside Alpheios for the neutral tone: `0` or `5`. The aforementioned code should, if necessary, be changed to work correctly with whatever values we'll agree upon.

Reply:
@kirlat , I have read again @balmas comment - that we use the same format method for mandarin and piniyn - that's why we it has this array of tones.
And it seems to me that - we shouldn't change anything here?
what do you think?

Reply:
@irina060981, I'm not sure myself, will wait for comment from @balmas. I guess if we decide to use `0` for the neutral tone and during dictionary conversion to our format we'll be replacing all `5`s with `0`s then there is nothing to change here. Buy if we decide to go with `5`s then this part will probably needs to be changed. Don't know how critical is that now, though.

Reply:
ok - let's wait for Bridget !

Reply:
sorry @irina060981 and @kirlat I think my response was confusing. pin yin is the romanization of the mandarin pronunciation, so they aren't the same thing but the formatting would apply equally to each.  

Regarding the 0 and the 5 tone -- I believe these are the same and should be formatted the same. Looking more closely at the code, I think that's in effect what is happening here anyway, because we we treat ANY number that isn't 1,2,3 or 4 as the neutral tone.  I do think this code could be made more explicit though.